internal/webhooks/cluster_test.go

```diff
@@ -409,6 +409,128 @@ func TestClusterValidation(t *testing.T) {
+					WithClusterNetwork(&clusterv1.ClusterNetwork{
+						Services: &clusterv1.NetworkRanges{
+							CIDRBlocks: []string{"2004::1234:abcd:ffff:c0a8:101/64"}},
+						Pods: &clusterv1.NetworkRanges{
+							CIDRBlocks: []string{"2004::1234:abcd:ffff:c0a8:101/64"}},
+					}).
+					Build(),
+			},
+			{
+				name:      "pass with valid dualstack CIDR ranges",
+				expectErr: false,
+				in: builder.Cluster("fooNamespace", "cluster1").
+					WithClusterNetwork(&clusterv1.ClusterNetwork{
+						Services: &clusterv1.NetworkRanges{
+							CIDRBlocks: []string{"2004::1234:abcd:ffff:c0a8:101/64", "10.10.10.10/24"}},
+						Pods: &clusterv1.NetworkRanges{
+							CIDRBlocks: []string{"2004::1234:abcd:ffff:c0a8:101/64", "10.10.10.10/24"}},
+					}).
+					Build(),
+			},
+			{
+				name:      "fails if multiple CIDR ranges of IPv4 are passed",
+				expectErr: true,
+				in: builder.Cluster("fooNamespace", "cluster1").
+					WithClusterNetwork(&clusterv1.ClusterNetwork{
+						Services: &clusterv1.NetworkRanges{
+							CIDRBlocks: []string{"10.10.10.10/24", "11.11.11.11/24"}},
+					}).
+					Build(),
+			},
+			{
+				name:      "fails if multiple CIDR ranges of IPv6 are passed",
+				expectErr: true,
+				in: builder.Cluster("fooNamespace", "cluster1").
+					WithClusterNetwork(&clusterv1.ClusterNetwork{
+						Services: &clusterv1.NetworkRanges{
+							CIDRBlocks: []string{"2002::1234:abcd:ffff:c0a8:101/64", "2004::1234:abcd:ffff:c0a8:101/64"}},
+					}).
+					Build(),
+			},
+			{
+				name:      "fails if too many cidr ranges are specified in the clusterNetwork pods field",
+				expectErr: true,
+				in: builder.Cluster("fooNamespace", "cluster1").
+					WithClusterNetwork(&clusterv1.ClusterNetwork{
+						Pods: &clusterv1.NetworkRanges{
+							CIDRBlocks: []string{"10.10.10.10/24", "11.11.11.11/24", "12.12.12.12/24"}}}).
+					Build(),
+			},
+			{
+				name:      "fails if too many cidr ranges are specified in the clusterNetwork pods field",
+				expectErr: true,
+				in: builder.Cluster("fooNamespace", "cluster1").
+					WithClusterNetwork(&clusterv1.ClusterNetwork{
+						Services: &clusterv1.NetworkRanges{
+							CIDRBlocks: []string{"10.10.10.10/24", "11.11.11.11/24", "12.12.12.12/24"}}}).
+					Build(),
+			},
+			{
+				name:      "fails if cidr ranges are not valid",
+				expectErr: true,
+				in: builder.Cluster("fooNamespace", "cluster1").
+					WithClusterNetwork(&clusterv1.ClusterNetwork{
+						Services: &clusterv1.NetworkRanges{
+							// Invalid ranges: missing network prefix
+							CIDRBlocks: []string{"10.10.10.10", "11.11.11.11"}}}).
+					Build(),
+			},
+			{
+				name:      "fails if pod and service cidr families do not match",
+				expectErr: true,
+				in: builder.Cluster("fooNamespace", "cluster1").
+					WithClusterNetwork(&clusterv1.ClusterNetwork{
+						// Services are IPv4
+						Services: &clusterv1.NetworkRanges{
+							CIDRBlocks: []string{"10.10.10.10/24", "11.11.11.11/24"}},
```

Comment:
```suggestion
							CIDRBlocks: []string{"10.10.10.10/24"}},
```
I think otherwise you don't get to the IPFamily code